src/api/cubrid_log.c

```diff
@@ -599,10 +610,45 @@ cubrid_log_connect_server_internal (char *host, int port, char *dbname)
 
   reason = ntohl (*(int *) recv_data);
 
+#if defined (WINDOWS)
+  if (reason == SERVER_CONNECTED_NEW)
+    {
+      css_queue_user_data_buffer (g_conn_entry, rid, sizeof (int), (char *) &reason);
+
+      if (css_receive_data (g_conn_entry, rid, &recv_data, &recv_data_size, g_connection_timeout * 1000) != NO_ERRORS)
+	{
+	  CUBRID_LOG_ERROR_HANDLING (CUBRID_LOG_FAILED_CONNECT,
+				     "Failed to receive the server port id from the master.\n");
+	}
+
+      if (recv_data != NULL && recv_data_size == sizeof (int))
+	{
+	  int port_id = ntohl (*((int *) recv_data));
+	  css_close_conn (g_conn_entry);
+
+	  if (recv_data != (char *) &reason)
```

Comment:
`reason` is a local variable and d`recv_data` seems to be allocated in a receive function. How do their addresses are the same? and let me know what it means if possible.

Reply:
Oh, then this shouldn't be free. Thank you for your question.

Reply:
If ,for some reason, the buffer (`&reason`) stored in the queue cannot be retrieved, free is necessary because internally malloc is performed on `recv_data`.

Reply:
Entry is created with the key (request id : `rid`) input from css_queue_user_data_buffer (), and store the buffer in the entry.
If there is no buffer matching request_id (rid) in the queue internally, memory is allocated and returned. 
Since it may not be possible to create an entry to store the buffer inside css_queue_user_data_buffer(), it seems appropriate to free it here as defense code.